### DISCUSSION TITLE
How to use factory calibration data (Shelly Plus 1PM)

### DISCUSSION THREAD

**Discussion by Tobi266** [arendst/Tasmota#18678] [Q&A] [answered]
Hi,

I flashed a Shelly Plus 1PM with Tasmota OTA. 
Guide: https://github.com/tasmota/mgos32-to-tasmota32

I extracted the shelly.bin with the factory calibrations.
![calibration](https://github.com/arendst/Tasmota/assets/84711635/c01f4a2e-5d7d-4785-8470-bc842bc232a8)

But how can I add/use these in Tasmota?

Thanks

**Comment by arendst** [comment, marked as answer]
Currently you cannot. Apparently shelly changed their calib data layout to trick tasmota. "someone" needs to find out how the new calib data relates to the old data.

Until then use local calibration as any other power monitoring device.

**Comment by palto42** [comment]
@arendst  Would it help to report the factory calibration values plus the Tasmota values after manual calibration here?
Maybe we can find patter for a conversion rule if we have a couple of samples?
